### DISCUSSION TITLE
Updating a template on a running workspace with coder_script changes.

### DISCUSSION THREAD

**Discussion by carlos-dc** [coder/coder#13525] [General]
So here is what I have experienced, I have a handful of templates that I have had to update by modifying a coder_script resource for adding/removing dependencies, adjusting building or deployment of internal repos, etc. When I start a workspace for the first time it runs just fine and the agent executes the coder_script. Same when I restart a workspace. The problem is while a workspace is running. When I push changes to a template by changing a coder_script and nothing else, the workspace prompts me that an update is available, I click the update button and the template version on the workspace gets changed but the script changes never execute on the running workspace. This requires me to first click the update to bump the template version, then full restart the workspace so that the script re-executes. 

Does anyone have any suggestions on how I can get the agent to recognize that the script changed and re-run it without a full restart?

I should note that if I make a change the coder_agent resource itself then terraform properly replaces it and it runs everything as expected, but changes to the agent itself are very rare in my case.

**Comment by carlos-dc** [comment]
Ok so I want to make an update, I missed the fact that the behavior when updating the coder_agent is actually an "Update and Restart" instead of just an update in place. I guess my question is a little different now, how can I force this behavior when I update my coder_scripts?

**Reply by stirby** [reply]
We have an enterprise feature that forces workspaces to stay on the latest template version ([Docs here](https://coder.com/docs/templates/general-settings#require-automatic-updates-enterprise)). When enabled, all workspaces will automatically update on the next restart whenever you publish a new template version. "Start and Update" becomes the only start option for those workspaces.

Does this help?
